diff -ruN ../mwm/scripts/events/event.sh.gentoo.in ./scripts/events/event.sh.gentoo.in
--- ../mwm/scripts/events/event.sh.gentoo.in	1970-01-01 03:00:00.000000000 +0300
+++ ./scripts/events/event.sh.gentoo.in	2009-05-20 00:23:01.000000000 +0400
@@ -0,0 +1,20 @@
+#!/bin/sh
+# Usage: event.sh <event> <if-name>
+
+case "$1" in
+    if-create)
+    ;;
+    if-up)
+        test -x /etc/init.d/net.wimax0 && /etc/init.d/net.wimax0 start
+    ;;
+    if-down)
+        test -x /etc/init.d/net.wimax0 && /etc/init.d/net.wimax0 stop
+    ;;
+    if-release)
+    ;;
+    *)
+	echo "Usage: $0 { if-create | if-up | if-down | if-release }" >&2
+	exit 3
+    ;;
+esac
+
diff -ruN ../mwm/scripts/events/Makefile.in ./scripts/events/Makefile.in
--- ../mwm/scripts/events/Makefile.in	2009-05-20 00:26:22.000000000 +0400
+++ ./scripts/events/Makefile.in	2009-05-20 00:32:19.000000000 +0400
@@ -152,7 +152,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 udevdir = @udevdir@
-@WITH_SCRIPT_TRUE@eventsdir = $(sysconfdir)
+@WITH_SCRIPT_TRUE@eventsdir = $(sysconfdir)/madwimax
 @WITH_SCRIPT_TRUE@events_SCRIPTS = event.sh
 @WITH_SCRIPT_TRUE@MOSTLYCLEANFILES = event.sh
 @WITH_SCRIPT_TRUE@DISTCLEANFILES = event.sh.$(script)
diff -ruN ../mwm/scripts/udev/z60_madwimax.rules.in ./scripts/udev/z60_madwimax.rules.in
--- ../mwm/scripts/udev/z60_madwimax.rules.in	2009-05-20 00:26:29.000000000 +0400
+++ ./scripts/udev/z60_madwimax.rules.in	2009-05-20 00:19:13.000000000 +0400
@@ -4,7 +4,7 @@
 
 ATTRS{idVendor}=="04e8", ATTRS{idProduct}=="6761", ATTR{bInterfaceNumber}=="01", RUN+="/bin/sh -c '/bin/echo -n $kernel > /sys/$devpath/driver/unbind'"
 ATTRS{idVendor}=="04e9", ATTRS{idProduct}=="6761", ATTR{bInterfaceNumber}=="01", RUN+="/bin/sh -c '/bin/echo -n $kernel > /sys/$devpath/driver/unbind'"
-#ATTR{idVendor}=="04e8", ATTR{idProduct}=="6761", RUN+="@SBINDIR@/madwimax -qd --exact-device=$attr{busnum}/$attr{devnum}"
-#ATTR{idVendor}=="04e9", ATTR{idProduct}=="6761", RUN+="@SBINDIR@/madwimax -qd --exact-device=$attr{busnum}/$attr{devnum}"
+ATTRS{idVendor}=="04e8", ATTRS{idProduct}=="6761", RUN+="@SBINDIR@/madwimax -qd --exact-device=$attr{busnum}/$attr{devnum}"
+ATTRS{idVendor}=="04e9", ATTRS{idProduct}=="6761", RUN+="@SBINDIR@/madwimax -qd --exact-device=$attr{busnum}/$attr{devnum}"
 
 LABEL="madwimax_rules_end"
diff -ruN ../mwm/src/wimax.c ./src/wimax.c
--- ../mwm/src/wimax.c	2009-05-20 00:26:29.000000000 +0400
+++ ./src/wimax.c	2009-05-20 00:24:27.000000000 +0400
@@ -245,7 +245,7 @@
 	} else if (pid > 0) { /* parent */
 		return pid;
 	} else { /* child */
-		char *args[] = {SYSCONFDIR "/event.sh", event, tap_dev, NULL};
+		char *args[] = {SYSCONFDIR "/madwimax/event.sh", event, tap_dev, NULL};
 		char *env[1] = {NULL};
 		/* run the program */
 		execve(args[0], args, env);
