--- dropwatch-1.4/src/Makefile.orig	2012-05-31 02:35:40.000000000 +0400
+++ dropwatch-1.4/src/Makefile	2012-10-14 00:35:10.791085211 +0400
@@ -1,11 +1,11 @@
 all: dropwatch
-CFLAGS+=-c -g -D_GNU_SOURCE -Wall -Werror `pkg-config --cflags libnl-3.0`
-LDFLAGS=-lbfd -lreadline -lnl-3 -lnl-genl-3
+CFLAGS+=-c -D_GNU_SOURCE -Wall -Werror `pkg-config --cflags libnl-3.0`
+LIBS=-lbfd -lreadline -lnl-3 -lnl-genl-3
 OBJFILES := main.o lookup.o\
 	 lookup_bfd.o lookup_kas.o
 
 dropwatch: $(OBJFILES) 
-	gcc -g -o dropwatch $(OBJFILES) $(LDFLAGS) 
+	gcc -o dropwatch $(LDFLAGS) $(OBJFILES) $(LIBS)
 
 %.o: %.c
 	gcc $(CFLAGS) $<
