--- maui-3.3.1/src/moab/MSched.c.orig	2011-03-04 19:28:25.000000000 +0300
+++ maui-3.3.1/src/moab/MSched.c	2012-05-12 06:09:25.504601149 +0400
@@ -4089,9 +4089,6 @@
 
       /* build stat buffer */
 
-      sprintf(Buf,"%ld\n",
-        S->Time);
-
       /* set up scheduler run time */
 
       RunTime = MStat.SchedRunTime;
@@ -4099,10 +4096,10 @@
       GP = &MPar[0];
       T  = &GP->S;
 
-      /*            STM ITM RTM IJ EJ AJ UN UP UM IN IP IM CT SJ TPA TPB SPH TMA TMD QP AQP NJA JAC PSX IT RPI WEF WI MXF ABP MBP AQT MQT PSR PSD PSU MSA MSD JE */
+      /*            CTM STM ITM RTM IJ EJ AJ UN UP UM IN IP IM CT SJ TPA TPB SPH TMA TMD QP AQP NJA JAC PSX IT RPI WEF WI MXF ABP MBP AQT MQT PSR PSD PSU MSA MSD JE */
 
-      sprintf(Buf,"%s%ld %ld %ld %d %d %d %d %d %d %d %d %d %d %d %lf %lf %lf %lf %lf %d %lu %lf %lf %lf %d %ld %lf %d %lf %lf %d %lf %lu %lf %lf %lf %lf %lf %d %d %lf\n",
-        Buf,
+      sprintf(Buf,"%ld\n%ld %ld %ld %d %d %d %d %d %d %d %d %d %d %d %lf %lf %lf %lf %lf %d %lu %lf %lf %lf %d %ld %lf %d %lf %lf %d %lf %lu %lf %lf %lf %lf %lf %d %d %lf\n",
+        S->Time,
         S->StartTime,
         MStat.InitTime,
         RunTime,
