--- latex2rtf-2.3.5/Makefile.orig	2013-11-12 12:44:44.000000000 +0400
+++ latex2rtf-2.3.5/Makefile	2014-02-24 17:55:03.068190671 +0400
@@ -217,7 +217,6 @@
 	cp -p doc/latex2rtf.1    $(DESTDIR)$(MANDIR)
 	cp -p doc/latex2png.1    $(DESTDIR)$(MANDIR)
 	cp -p $(CFGS)            $(DESTDIR)$(CFGDIR)
-	cp -p doc/latex2rtf.html $(DESTDIR)$(SUPPORTDIR)
 	cp -p doc/latex2rtf.pdf  $(DESTDIR)$(SUPPORTDIR)
 	cp -p doc/latex2rtf.txt  $(DESTDIR)$(SUPPORTDIR)
 	cp -p $(BINARY_NAME)     $(DESTDIR)$(BINDIR)
