--- hydra-7.4.2/configure.orig	2013-08-26 07:08:18.000000000 +0400
+++ hydra-7.4.2/configure	2013-08-26 07:16:00.252186169 +0400
@@ -195,12 +195,6 @@
         fi
     fi
     if [ "X" = "X$CURSES_PATH" ]; then
-        TMP_LIB=`/bin/ls $i/libcurses.so.* 2> /dev/null | grep curses.`
-        if [ -n "$TMP_LIB" ]; then
-            CURSES_PATH="$i"
-        fi
-    fi
-    if [ "X" = "X$CURSES_PATH" ]; then
         TMP_LIB=`/bin/ls $i/libcurses.dll* 2> /dev/null | grep curses.`
         if [ -n "$TMP_LIB" ]; then
             CURSES_PATH="$i"
@@ -392,12 +386,6 @@
         fi
     fi
     if [ "X" = "X$FIREBIRD_PATH" ]; then
-        TMP_LIB=`/bin/ls $i/libfbclient.so.* 2> /dev/null | grep libfbclient.`
-        if [ -n "$TMP_LIB" ]; then
-          FIREBIRD_PATH="$i"
-        fi
-    fi
-    if [ "X" = "X$FIREBIRD_PATH" ]; then
         TMP_LIB=`/bin/ls $i/libfbclient.dll* 2> /dev/null | grep libfbclient.`
         if [ -n "$TMP_LIB" ]; then
           FIREBIRD_PATH="$i"
@@ -428,12 +416,6 @@
         fi
     fi
     if [ "X" = "X$MYSQL_PATH" ]; then
-        TMP_LIB=`/bin/ls $i/libmysqlclient.so.* 2> /dev/null | grep libmysqlclient.`
-        if [ -n "$TMP_LIB" ]; then
-          MYSQL_PATH="$i"
-        fi
-    fi
-    if [ "X" = "X$MYSQL_PATH" ]; then
         TMP_LIB=`/bin/ls $i/libmysqlclient.dll* 2> /dev/null | grep libmysqlclient.`
         if [ -n "$TMP_LIB" ]; then
           MYSQL_PATH="$i"
@@ -468,12 +450,6 @@
         fi
     fi
     if [ "X" = "X$AFP_PATH" ]; then
-        TMP_LIB=`/bin/ls $i/libafpclient.so.* 2> /dev/null | grep libafpclient.`
-        if [ -n "$TMP_LIB" ]; then
-          AFP_PATH="$i"
-        fi
-    fi
-    if [ "X" = "X$AFP_PATH" ]; then
         TMP_LIB=`/bin/ls $i/libafpclient.dll* 2> /dev/null | grep libafpclient.`
         if [ -n "$TMP_LIB" ]; then
           AFP_PATH="$i"
@@ -504,12 +480,6 @@
         fi
     fi
     if [ "X" = "X$NCP_PATH" ]; then
-        TMP_LIB=`/bin/ls $i/libncp.so.* 2> /dev/null | grep ncp.`
-        if [ -n "$TMP_LIB" ]; then
-            NCP_PATH="$i"
-        fi
-    fi
-    if [ "X" = "X$NCP_PATH" ]; then
         TMP_LIB=`/bin/ls $i/libncp.dll* 2> /dev/null | grep ncp.`
         if [ -n "$TMP_LIB" ]; then
             NCP_PATH="$i"
@@ -678,18 +648,6 @@
         fi
     fi
     if [ "X" = "X$ORACLE_PATH" ]; then
-        TMP_LIB=`/bin/ls $i/libocci.so.* 2> /dev/null | grep occi.`
-        if [ -n "$TMP_LIB" ]; then
-            ORACLE_PATH="$i"
-        fi
-        if [ "X" != "X$ORACLE_PATH" ]; then
-          TMP_LIB=`/bin/ls $i/libclntsh.so.* 2> /dev/null | grep clntsh.`
-          if [ -z "$TMP_LIB" ]; then
-            ORACLE_PATH=""
-          fi
-        fi
-    fi
-    if [ "X" = "X$ORACLE_PATH" ]; then
         TMP_LIB=`/bin/ls $i/libocci.dll* 2> /dev/null | grep occi.`
         if [ -n "$TMP_LIB" ]; then
             ORACLE_PATH="$i"
@@ -711,16 +669,6 @@
             LIBAIO="$i"
         fi
      fi
-     if [ "X" = "X$LIBAIO" ]; then
-            TMP_LIB=`/bin/ls $i/libaio.so.* 2> /dev/null | grep aio.`
-            if [ -n "$TMP_LIB" ]; then
-              LIBAIO="$i"
-            fi
-            TMP_LIB=`/bin/ls $i/libaio.dll* 2> /dev/null | grep aio.`
-            if [ -n "$TMP_LIB" ]; then
-              LIBAIO="$i"
-            fi
-     fi
   done
   if [ "X" = "X$LIBAIO" ]; then
     ORACLE_PATH=""
