diff -Naur a/configure b/configure
--- a/configure	2013-08-02 07:35:56.000000000 -0400
+++ b/configure	2013-08-27 13:26:37.386362215 -0400
@@ -79,9 +79,11 @@
 SOCKET_PATH=""
 MANDIR=""
 XHYDRA_SUPPORT=""
-LIBDIRS=`cat /etc/ld.so.conf /etc/ld.so.conf.d/* 2> /dev/null | grep -v '^#' | sort | uniq`
 if [ "$SIXFOUR" = "64" ]; then
+  LIBDIRS=`cat /etc/ld.so.conf /etc/ld.so.conf.d/* 2> /dev/null | grep -v '^#' | grep -v 'lib32' | sort | uniq`
   LIBDIRS="$LIBDIRS /lib64 /usr/lib64 /usr/local/lib64 /opt/local/lib64"
+else
+  LIBDIRS=`cat /etc/ld.so.conf /etc/ld.so.conf.d/* 2> /dev/null | grep -v '^#' | sort | uniq`
 fi
 LIBDIRS="$LIBDIRS /lib /usr/lib /usr/local/lib /opt/local/lib"
 INCDIRS="/usr/include /usr/local/include /opt/include /opt/local/include"
