diff -Naurd chromium-22.0.1229.14/third_party/qcms/src.orig/chain.c chromium-22.0.1229.14/third_party/qcms/src/chain.c
--- chromium-22.0.1229.14/third_party/qcms/src.orig/chain.c	2012-08-22 11:01:52.000000000 +0400
+++ chromium-22.0.1229.14/third_party/qcms/src/chain.c	2012-08-28 23:17:17.875806163 +0400
@@ -29,6 +29,8 @@
 #include "transform_util.h"
 #include "matrix.h"
 
+#pragma GCC target ("no-sse2")
+
 static struct matrix build_lut_matrix(struct lutType *lut)
 {
 	struct matrix result;
diff -Naurd chromium-22.0.1229.14/third_party/qcms/src.orig/iccread.c chromium-22.0.1229.14/third_party/qcms/src/iccread.c
--- chromium-22.0.1229.14/third_party/qcms/src.orig/iccread.c	2012-08-22 11:01:52.000000000 +0400
+++ chromium-22.0.1229.14/third_party/qcms/src/iccread.c	2012-08-28 23:17:39.328221658 +0400
@@ -27,6 +27,8 @@
 #include <string.h> //memset
 #include "qcmsint.h"
 
+#pragma GCC target ("no-sse2")
+
 /* It might be worth having a unified limit on content controlled
  * allocation per profile. This would remove the need for many
  * of the arbitrary limits that we used */
diff -Naurd chromium-22.0.1229.14/third_party/qcms/src.orig/matrix.c chromium-22.0.1229.14/third_party/qcms/src/matrix.c
--- chromium-22.0.1229.14/third_party/qcms/src.orig/matrix.c	2012-08-22 11:01:52.000000000 +0400
+++ chromium-22.0.1229.14/third_party/qcms/src/matrix.c	2012-08-28 23:17:47.427875178 +0400
@@ -25,6 +25,8 @@
 #include "qcmsint.h"
 #include "matrix.h"
 
+#pragma GCC target ("no-sse2")
+
 struct vector matrix_eval(struct matrix mat, struct vector v)
 {
 	struct vector result;
diff -Naurd chromium-22.0.1229.14/third_party/qcms/src.orig/transform.c chromium-22.0.1229.14/third_party/qcms/src/transform.c
--- chromium-22.0.1229.14/third_party/qcms/src.orig/transform.c	2012-08-22 11:01:52.000000000 +0400
+++ chromium-22.0.1229.14/third_party/qcms/src/transform.c	2012-08-28 23:18:08.290316230 +0400
@@ -30,6 +30,8 @@
 #include "matrix.h"
 #include "transform_util.h"
 
+#pragma GCC target ("no-sse2")
+
 /* for MSVC, GCC, Intel, and Sun compilers */
 #if defined(_M_IX86) || defined(__i386__) || defined(__i386) || defined(_M_AMD64) || defined(__x86_64__) || defined(__x86_64)
 #define X86
diff -Naurd chromium-22.0.1229.14/third_party/qcms/src.orig/transform_util.c chromium-22.0.1229.14/third_party/qcms/src/transform_util.c
--- chromium-22.0.1229.14/third_party/qcms/src.orig/transform_util.c	2012-08-22 11:01:52.000000000 +0400
+++ chromium-22.0.1229.14/third_party/qcms/src/transform_util.c	2012-08-28 23:18:16.523297506 +0400
@@ -28,6 +28,8 @@
 #include "transform_util.h"
 #include "matrix.h"
 
+#pragma GCC target ("no-sse2")
+
 #if !defined(INFINITY)
 #define INFINITY HUGE_VAL
 #endif
