diff -Naur bind2nsd-0.7.2.orig/bind2nsd/Config.py bind2nsd-0.7.2/bind2nsd/Config.py
--- bind2nsd-0.7.2.orig/bind2nsd/Config.py	2008-06-27 04:22:28.000000000 +0400
+++ bind2nsd-0.7.2/bind2nsd/Config.py	2010-12-07 12:09:19.000000000 +0300
@@ -70,7 +70,7 @@
 	   'masters_for_slaves'	 : '127.0.0.1',
 	   'named-checkconf'     : '/usr/sbin/named-checkconf',
 	   'named-checkzone'     : '/usr/sbin/named-checkzone',
-	   'named_root'          : '/etc/bind9',
+	   'named_root'          : '/etc/bind',
            'named_conf'          : 'named.conf',
 	   'named_watchlist'     : '/etc/named.conf',
 	   'nsd-checkconf'       : '/usr/sbin/nsd-checkconf',
diff -Naur bind2nsd-0.7.2.orig/etc/bind2nsd.conf bind2nsd-0.7.2/etc/bind2nsd.conf
--- bind2nsd-0.7.2.orig/etc/bind2nsd.conf	2008-06-27 04:22:28.000000000 +0400
+++ bind2nsd-0.7.2/etc/bind2nsd.conf	2010-12-07 12:09:05.000000000 +0300
@@ -34,7 +34,7 @@
 
 #--- parameters for the bind2nsd command ---------------------------------
 
-# default: named_root = /etc/bind9
+# default: named_root = /etc/bind
 # default: named_conf = named.conf
 # default: nsd_conf = nsd.conf
 # default: nsd_conf_dir = /etc/nsd/
@@ -72,7 +72,7 @@
 # default: dnspw = iforgot => dnsconfig password (see s64-mkpw)
 # default: named-checkconf = /usr/sbin/named-checkconf => BIND command location
 # default: named-checkzone = /usr/sbin/named-checkzone => BIND command location
-# default: named_root = /etc/bind9 => default BIND config directory
+# default: named_root = /etc/bind => default BIND config directory
 # default: named_conf = named.conf => BIND config file name
 # default: named_watchlist = /etc/named.conf => config files to poll
 # default: password_file = /etc/bind2nsd/passwd => encrypted passwords
