--- Makefile.in.orig	2012-08-30 13:41:32.000000000 +0400
+++ Makefile.in	2012-08-30 23:55:17.000000000 +0400
@@ -856,13 +856,13 @@
 	$(E)ar rcs $@ $(ULIBTHREADEDOBJS)
 
 # rule for building the shared ofs library
-lib/libofs.so: $(ULIBPICOBJS)
+lib/libofs.so: $(ULIBPICOBJS) lib/libpvfs2.so
 	$(Q) "  LDSO		$@"
 	$(E)$(INSTALL) -d lib
 	$(E)$(LDSHARED) -Wl,-soname,libofs.so -o $@ $(ULIBPICOBJS) $(ULIBDEPLIBS) $(DEPLIBS)
 
 # rule for building the shared ofs _multithreaded_ library
-lib/libofs-threaded.so: $(ULIBTHREADEDPICOBJS)
+lib/libofs-threaded.so: $(ULIBTHREADEDPICOBJS) lib/libpvfs2.so
 	$(Q) "  LDSO		$@"
 	$(E)$(INSTALL) -d lib
 	$(E)$(LDSHARED) -Wl,-soname,libofs-threaded.so -o $@ $(ULIBTHREADEDPICOBJS) $(ULIBDEPLIBS) $(DEPLIBS)
