--- src/metalog.c.orig	2011-01-24 05:14:55.000000000 +0300
+++ src/metalog.c	2012-04-26 14:29:02.108312392 +0400
@@ -205,6 +205,7 @@
             new_output->maxsize = (*cur_block)->maxsize;
             new_output->maxfiles = (*cur_block)->maxfiles;
             new_output->maxtime = (*cur_block)->maxtime;
+            new_output->postrotate_cmd = (*cur_block)->postrotate_cmd;
             new_output->showrepeats = (*cur_block)->showrepeats;
             new_output->stamp_fmt = (*cur_block)->stamp_fmt;
             new_output->flush = (*cur_block)->flush;
